pkg/extractor/regex_extractor.go

```diff
@@ -22,8 +21,6 @@ func NewRegexUrlExtractor() (*RegexUrlExtractor, error) {
 // Extract implements the UrlExtractor interface, using the regex to find urls in the given text.
 func (re *RegexUrlExtractor) Extract(text string) []string {
 	matches := re.extractor.FindAllString(text, -1)
-	for i, match := range matches {
-		matches[i] = strings.ToLower(match)
-	}
+	// The copy step is unnecessary here; you can directly return matches
```

Comment:
Let's do
`return matches` OR
 `return re.extractor.FindAllString(text, -1)` and remove the unnecessary comment.

Reply:
Comment is removed